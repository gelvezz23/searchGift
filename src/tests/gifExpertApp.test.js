/** @format */

import React from 'react';

import { shallow } from 'enzyme';
import GifExpertApp from './../gifExpertApp';

describe('Pruebas en mi componente <gifExpertApp/>', () => {
	const wrapper = shallow(<GifExpertApp />);

	test('debe de visualizarse correctamente', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('debe mostrat una lista de categorias ', () => {
		const categories = ['hunter x hunter', 'dragon ball'];
		const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('GifGrid').length).toBe(categories.length);
	});
});
