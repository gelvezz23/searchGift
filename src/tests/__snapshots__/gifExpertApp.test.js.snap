// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en mi componente <gifExpertApp/> debe de visualizarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    categories={Array []}
    setcategories={[Function]}
  />
  <hr />
</Fragment>
`;

exports[`Pruebas en mi componente <gifExpertApp/> debe mostrat una lista de categorias  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    categories={
      Array [
        "hunter x hunter",
        "dragon ball",
      ]
    }
    setcategories={[Function]}
  />
  <hr />
  <GifGrid
    category="hunter x hunter"
    key="0"
  />
  <GifGrid
    category="dragon ball"
    key="1"
  />
</Fragment>
`;
